lib/Sema/ConstraintSystem.cpp

```diff
@@ -3743,6 +3747,14 @@ static bool diagnoseConflictingGenericArguments(ConstraintSystem &cs,
     llvm::SmallSetVector<Type, 4> arguments;
     for (const auto &solution : solutions) {
       auto type = solution.typeBindings.lookup(typeVar);
+      // Type variables gathered from a solution's type binding context may not
+      // exist in another given solution because each solution could have a
+      // different set of overload choices picked, which implies that some of
+      // the generic parameters, and type variables that represent them, would
+      // be unique to that solution.
```

Comment:
I don't think it's due to differences in overloads, as the check at the start of this function checks for an empty overload diff set. From a quick look into it, it seems to be due to type binding logic introducing a new type variable when opening the `Element` parameter type for one of the solutions:

https://github.com/apple/swift/blob/4bef7bf089846e4c7c04fb361dbab006c3dabe64/lib/Sema/CSBindings.cpp#L1810-L1825

Reply:
I don't think it needs to enumerate every possible situation where a new type variable could be introduced for a generic parameter e.g. bound and unbound generic types, keypath etc. just needs to mention a most common circumstance. Maybe we should change it to `different set of overload and/or generic type choices ...`.

Reply:
Right yeah, I really just meant that it's misleading to talk about different overloads as this bit of diagnostic logic runs only on ambiguities that aren't a result of different overload choices. Personally I'd be fine with a more vague `some solutions may have additional type variables not present in other solutions due to taking different paths in the solver`.

Reply:
Sounds good to me! @LucianoPAlmeida mind adjusting the comment?

Reply:
Sure!

Reply:
Thank you!